plugin/core/views.py

```diff
@@ -172,3 +175,42 @@ def text_document_range_formatting(view: sublime.View, region: sublime.Region) -
         "options": formatting_options(view.settings()),
         "range": region_to_range(view, region).to_lsp()
     })
+
+
+def minihtml(content: Optional[Union[str, dict]], view: sublime.View) -> str:
+    """ Content can be a string or MarkupContent """
+    if isinstance(content, str):
+        return text2html(content)
+    elif isinstance(content, dict):
+        value = content.get("value") or ""
+        kind = content.get("kind")
+        if kind == "markdown":
+            return mdpopups.md2html(view, value)
+        else:
+            # must be plaintext
+            return text2html(value)
+    else:
+        return ''
+
+
+def text2html(content: str) -> str:
+    content = html.escape(content).replace('\n', '<br>')
+
+    def repalce_url_with_link(matchObj: Any) -> str:
+        href_tag, url = matchObj.groups()
+        if href_tag:
```

Comment:
This can never happen right? HTML is already escaped, so we will never match `<a href="`. Please add a test for this in test_views.py if I'm wrong.

Reply:
I removed the regex for finding `<a href` because we already escaped the string. Instead I  think we should only find URLs and make them clickable.